Makefile

```diff
@@ -322,6 +281,9 @@ localunit: test/goecho/goecho varlink_generate
 	$(GO) tool cover -func=${COVERAGE_PATH}/coverprofile > ${COVERAGE_PATH}/functions
 	cat ${COVERAGE_PATH}/functions | sed -n 's/\(total:\).*\([0-9][0-9].[0-9]\)/\1 \2/p'
 
+.PHONY: test
+test: localunit localintegration remoteintegration localsystem remotesystem  ## Run unit, integration, and system tests.
```

Comment:
I wonder if `make` will try to parallelize these? That would make for a complete disaster. (I'm not going to even try it, because half of the above tests fail catastrophically on my laptop). Anyhow, I get the feeling that even without parallel execution this will not work for many people.

This is tough. I really like the idea of `make test`, it's just the right thing to do, but I have no way to actually verify or write such a thing.

Reply:
Make does not parallelize by default, you have to pass `-j`.

I'm in the same boat, no way to see if this works.  I have the feeling that the number of people who actually use this target is so small that either:

* They will complain if it's broken
* They will fix it if it's broken
* Both :smile: